### PR TITLE
chore(dep): bump cranelift to 0.84.0

### PR DESCRIPTION
**Pull request by waynexia**
Signed-off-by: Ruihang Xia <waynestxia@gmail.com>

# Which issue does this PR close?

Closes #2592 
Closes #2593
Closes #2594
Closes #2595

 # Rationale for this change

Cannot find a formal release note. Got [this](https://github.com/bytecodealliance/wasmtime/blob/main/RELEASES.md#0370) from wasmtime 0.37 version's release:

>Updated Cranelift to use regalloc2, a new register allocator. This should result in ~20% faster compile times, and for programs that suffered from register-allocation pressure before, up to ~20% faster generated code.

# What changes are included in this PR?

# Are there any user-facing changes?

# Does this PR break compatibility with Ballista?

### CODE REVIEW

File: datafusion/common/Cargo.toml [hunk 1]
```diff
@@ -24,7 +24,7 @@ repository = "https://github.com/apache/arrow-datafusion"
 readme = "README.md"
 authors = ["Apache Arrow <dev@arrow.apache.org>"]
 license = "Apache-2.0"
-keywords = [ "arrow", "query", "sql" ]
+keywords = ["arrow", "query", "sql"]
```
Comment (waynexia) [inline datafusion/common/Cargo.toml:27]: this is unrelated and not intended :frowning:

**Review by yjshen — APPROVED** [review]
